Adding lead and lag

mjkarlsen opened the pull request:
Added lead and lag functionality to be consistent with tidy methods

markfairbanks commented:
Awesome, these look good.

Can you make 2 tests for each lead and lag? 1 that shows it works on a vector, and one that shows that it works inside of mutate? They can all be in the same test file.

mjkarlsen commented:
Added the tests for lead and lag. There was still an error with lag when trying to use usemethod but I fixed it by masking stats::lag similar to how it was done in dplyr.
